### PR TITLE
Adjusted logic for filtering zero-coverage samples and intervals in CreateReadCountPanelOfNormals.

### PR DESCRIPTION
**Pull request by samuelklee**
Just a minor fix, but could conceivably change results by keeping/dropping samples/intervals on the edge of the filter. See discussion in https://gatk.broadinstitute.org/hc/en-us/community/posts/360057785591-Error-while-running-CreateReadCountPanelOfNormals

### CODE REVIEW

**Comment by samuelklee** [comment]
@fleharty forgot about this short PR. Might be nice to get it in before release?

File: src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java
Comment (fleharty) [inline src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java:241]: I'm a little confused, why is numZerosInSample a double rather than an int?
If you need it to be a double so that the fraction is a double, why not cast at the point of computing the fraction?

**Review by fleharty — CHANGES_REQUESTED** [review]
@samuelklee Although a small change, can you add a simple test that passes with this change, and fails on previous code?

File: src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java
Comment (samuelklee) [reply src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java:241]: In both cases, the cast happens in the next line and the variable is not used elsewhere, so I'm OK keeping it like this. Certainly it's valid to represent an integer with a double, at least here...?

Comment (fleharty) [reply src/main/java/org/broadinstitute/hellbender/tools/copynumber/denoising/SVDDenoisingUtils.java:241]: @samuelklee It's fine the way it is.

**Comment by samuelklee** [comment]
Hmm, thanks for suggesting the addition of a regression test @fleharty. This caused me to realize that I actually missed another gap in the previous filtering logic that led to NaNs (resulting from division by zero interval medians) that occur in this particular edge case, which actually occurs before the rounding error I originally fixed. This leads to the erroneous noop filtering in the user's log:

```
15:45:24.835 INFO  SVDDenoisingUtils - Filtering intervals with median (across samples) less than or equal to the 10.00 percentile...
15:45:24.844 INFO  SVDDenoisingUtils - After filtering, 24745 out of 24745 intervals remain...
```

To prevent this scenario, I added another change to the filtering logic to always filter out such zero-median intervals---even if the interval-median filter is disabled. This will allow the user to actually build a PoN in this single-sample PoN edge case (rather than failing later---when attempting to perform the PCA with the NaNs present---with a message to use stricter filtering). But you may want to double check my logic to make sure I didn't miss anything else.

I added a regression test that only passes after the change. Unfortunately, because we use simulated data in the CreateReadCountPanelOfNormalsIntegrationTest that covers this code, it's a bit of a pain to add a test that covers regression of the rounding errors I originally set out to fix---but hopefully this is an acceptable compromise.

I think that all of this filtering logic was ported over from GATK CNV (I only rewrote the code to perform the filtering in-place to improve memory usage), and I'm not sure that all edge-case behavior was well defined by the original logic. Fortunately, these edge-case usages (i.e., using a single sample to build the PoN, mistakenly including too many uncovered intervals, and/or disabling various filters) are probably not too common.

Give me another day to make sure that results don't change on typical data (I'll just rebuild a typical PoN and check that nothing changes) and maybe to check a few more edge cases by hand. If we don't merge before @droazen cuts the release, no biggie.

**Comment by samuelklee** [comment]
Hmm, actually, let me take a second look at this. I'm a bit confused by the no-op above and want to do some more testing of edge cases just to make sure I'm not missing anything. I also got a bit confused looking back at the original forum post by the fact that two different users were posting about slightly different scenarios.

Apologies, but it's been a while since I opened this, so I need to refresh my memory!

**Comment by samuelklee** [comment]
Fixed up a few more minor things, back to you, @fleharty!

**Comment by samuelklee** [comment]
Oops, forgot to set some random seeds and got failures on Travis that were passing locally. Think it should be OK now. Good looking out Travis RNGs, you da real MVPs.

**Comment by samuelklee** [comment]
@fleharty can we close this out?

**Comment by samuelklee** [comment]
@fleharty just a reminder about this PR, let's try to get it in before next release.

**Comment by fleharty** [comment]
I think this is great, :+1:

**Comment by samuelklee** [comment]
@fleharty thanks! I think I need an explicit approval to merge.

**Review by fleharty — APPROVED** [review]
Sorry,  I didn't realize that didn't realize I didn't approve.

**Pull request merged**
